### PR TITLE
added hubspotv2 scanner

### PR DESCRIPTION
**Pull request by ankushgoel27**
### Description:
This PR adds the new v2 version of the hubspot API key. The older version still works.
https://developers.hubspot.com/docs/api/settings/account-information-api

https://developers.hubspot.com/changelog/upcoming-api-key-sunset

https://knowledge.hubspot.com/integrations/how-do-i-get-my-hubspot-api-key

hubspotv2

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/hubspotapikeyv2/hubspotapikeyv2.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+type Scanner struct{}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	client = common.SaneHttpClient()
+
+	keyPat = regexp.MustCompile(`\b(pat-na1-[A-Za-z0-9]{8}\-[A-Za-z0-9]{4}\-[A-Za-z0-9]{4}\-[A-Za-z0-9]{4}\-[A-Za-z0-9]{12})\b`)
```
Comment (rgmz) [inline pkg/detectors/hubspotapikeyv2/hubspotapikeyv2.go:24]:
```
pat-na1-
```
I can't find official documentation for this, but it looks like there are region-specific prefixes including `pat-na1-` and `pat-eu1-`. It's possible there are others.

Comment (ankushgoel27) [reply pkg/detectors/hubspotapikeyv2/hubspotapikeyv2.go:27]: how did you find the pat-eu1 format? We can add these 2 formats for now and add more later?

Comment (ankushgoel27) [reply pkg/detectors/hubspotapikeyv2/hubspotapikeyv2.go:27]: any updates on this?

**Comment by rgmz** [comment]
@zricethezav Any chance this can get merged?

**Review by rgmz — COMMENTED** [review]
Another observation.

**Pull request closed without merging**